osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs

```diff
@@ -546,5 +546,23 @@ protected override void Dispose(bool isDisposing)
             if (metadataClient.IsNotNull())
                 metadataClient.MultiplayerRoomScoreSet -= onRoomScoreSet;
         }
+
+        [Resolved]
+        private OsuGame? game { get; set; }
+
+        public void PresentBeatmap(WorkingBeatmap beatmap, RulesetInfo ruleset)
+        {
+            if (!this.IsCurrentScreen())
+                return;
+
+            // We can only handle the current daily challenge beatmap.
+            // If the import was for a different beatmap, pass the duty off to global handling.
+            if (beatmap.BeatmapSetInfo.OnlineID != playlistItem.Beatmap.BeatmapSet!.OnlineID)
+            {
+                game?.PresentBeatmap(beatmap.BeatmapSetInfo, b => b.ID == beatmap.BeatmapInfo.ID);
```

Comment:
This is infinitely looping, creating millions of `PerformFromMenuRunner` classes and whatnot other garbage.